plugins/tedge_configuration_plugin/src/main.rs

```diff
@@ -0,0 +1,135 @@
+use clap::Parser;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tedge_actors::Runtime;
+use tedge_config::system_services::get_log_level;
+use tedge_config::system_services::set_log_level;
+use tedge_config::TEdgeConfig;
+use tedge_config::TEdgeConfigLocation;
+use tedge_config::TEdgeConfigRepository;
+use tedge_config::DEFAULT_TEDGE_CONFIG_PATH;
+use tedge_config_manager::ConfigManagerBuilder;
+use tedge_config_manager::ConfigManagerConfig;
+use tedge_config_manager::ConfigManagerOptions;
+use tedge_downloader_ext::DownloaderActor;
+use tedge_file_system_ext::FsWatchActorBuilder;
+use tedge_health_ext::HealthMonitorBuilder;
+use tedge_http_ext::HttpActor;
+use tedge_mqtt_ext::MqttActorBuilder;
+use tedge_signal_ext::SignalActor;
+
+const AFTER_HELP_TEXT: &str = r#"On start, `tedge-configuration-plugin` notifies of its 
+managed configuration files and sends this list via MQTT.
+`tedge-configuration-plugin` subscribes to the topics for the commands `config_snapshot`
+and `config_update`.
+
+The thin-edge `CONFIG_DIR` is used:
+* to find the `tedge.toml` where the following configs are defined:
+  ** `mqtt.bind.address` and `mqtt.bind.port` to connect to the tedge MQTT broker
+  ** `root.topic` and `device.topic`: for the MQTT topics to publish to and subscribe from
```

Comment:
Spec also needs to be changed accordingly. #2231.
```suggestion
  ** `mqtt.topic_root` and `mqtt.device_topic_id`: for the MQTT topics to publish to and subscribe from
```